src/Storages/StorageFactory.h

```diff
@@ -29,6 +29,10 @@ class StorageFactory : private boost::noncopyable, public IHints<>
 
     static StorageFactory & instance();
 
+    /// Helper function to validate if a specific storage supports a setting
+    /// Meant to validate if a table engine before query start
```

Comment:
What did you mean by `validate if a table engine before query start` here?

Reply:
I'll reword this properly. It's used to validate if settings are query or engine settings before start the query interpretation